### PR TITLE
feat: get image by name/region/servertype

### PR DESCRIPTION
**Pull request by DenBeke**

### CODE REVIEW

File: api/marketplace/v1/marketplace_utils.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import (
+	"fmt"
+	"strings"
+
+	"github.com/scaleway/scaleway-sdk-go/utils"
+)
+
+// GetImageByName search for an image with the given name (exact match) in the given region
+// it returns the latest version of this specific image.
+func (s *Api) GetImageByName(imageName string, zone utils.Zone, commercialType string) (string, error) {
```
Comment (jerome-quere) [inline api/marketplace/v1/marketplace_utils.go]: `FindLocalImageIdByName`

File: api/marketplace/v1/marketplace_utils_test.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+	"github.com/dnaeon/go-vcr/recorder"
+	"github.com/scaleway/scaleway-sdk-go/internal/testhelpers"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+	"github.com/scaleway/scaleway-sdk-go/utils"
+)
+
+func TestGetImageByName(t *testing.T) {
+
+	// Setup recorder and scw client
+	r, err := recorder.NewAsMode("fixtures/marketplace", recorder.ModeReplaying, nil)
```
Comment (jerome-quere) [inline api/marketplace/v1/marketplace_utils_test.go]: rename `fixtures/marketplace ` by `testdata/go-vcr`
```
$> go help test | grep data                                                                                                                                                    
The go tool will ignore a directory named "testdata", making it available
to hold ancillary data needed by the tests
```

File: utils/locality_parsing.go [hunk 3]
```diff
@@ -0,0 +1,64 @@
+package utils
```
Comment (jerome-quere) [inline utils/locality_parsing.go]: Could be kept in same file that locality

File: utils/locality_parsing.go [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+func ParseZone(zone string) Zone {
+	switch zone {
+	case "par1":
+		// LEGACY
+		return ZoneFrPar1
+	case "ams1":
+		// LEGACY
+		return ZoneNlAms1
+	default:
+		return Zone(zone)
```
Comment (jerome-quere) [inline utils/locality_parsing.go]: Add a comment
```
// TODO: Add a warning message when unknown zone
```

File: utils/locality_parsing.go [hunk 5]
```diff
@@ -0,0 +1,64 @@
+package utils
+
+import (
+	"encoding/json"
+)
+
+// ParseZone parse a string value into a Zone object
+func ParseZone(zone string) Zone {
```
Comment (jerome-quere) [inline utils/locality_parsing.go]: Could return an error for future use case and avoid breaking change

**Comment by DenBeke** [comment]
> Why don't you use request objects?
> 
> ```
> FindLocalImageIDByName(&FindLocalImageIDByNameRequest{})
> ```

Because it's not auto-generated code with a specific goal.

File: utils/locality.go [hunk 6]
```diff
@@ -41,3 +43,70 @@ func (r Region) GetZones() []Zone {
 		return []Zone{}
 	}
 }
+
+// ParseZone parse a string value into a Zone object
+func ParseZone(zone string) (Zone, error) {
+	switch zone {
+	case "par1":
+		// LEGACY
```
Comment (kindermoumoute) [inline utils/locality.go]: `logger.Warningf("par1 is a deprecated name for zone, use fr-par instead")`

File: utils/locality.go [hunk 7]
```diff
@@ -41,3 +43,70 @@ func (r Region) GetZones() []Zone {
...
+		return ZoneFrPar1, nil
+	case "ams1":
+		// LEGACY
+		return ZoneNlAms1, nil
```
Comment (kindermoumoute) [inline utils/locality.go]: `logger.Warningf("ams1 is a deprecated name for zone, use nl-ams instead")`

File: utils/locality.go [hunk 8]
```diff
@@ -41,3 +43,70 @@ func (r Region) GetZones() []Zone {
...
+		return err
+	}
+	return nil
+}
+
+// ParseRegion parse a string value into a Zone object
+func ParseRegion(region string) (Region, error) {
+	switch region {
+	case "par1":
+		// LEGACY
```
Comment (kindermoumoute) [inline utils/locality.go]: `logger.Warningf("par1 is a deprecated name for region, use fr-par instead")`

File: utils/locality.go [hunk 9]
```diff
@@ -41,3 +43,70 @@ func (r Region) GetZones() []Zone {
...
+		return RegionFrPar, nil
+	case "ams1":
+		// LEGACY
```
Comment (kindermoumoute) [inline utils/locality.go]: `logger.Warningf("ams1 is a deprecated name for region, use nl-ams instead")`

File: utils/locality.go [hunk 10]
```diff
@@ -41,3 +43,70 @@ func (r Region) GetZones() []Zone {
...
+		return RegionNlAms, nil
+	default:
+		return Region(region), nil
+		// TODO: Add a warning message when unknown zone
```
Comment (kindermoumoute) [inline utils/locality.go]: yes

File: utils/locality.go [hunk 11]
```diff
@@ -41,3 +43,70 @@ func (r Region) GetZones() []Zone {
...
+	switch zone {
+	case "par1":
+		// LEGACY
+		return ZoneFrPar1, nil
+	case "ams1":
+		// LEGACY
+		return ZoneNlAms1, nil
+	default:
+		return Zone(zone), nil
+		// TODO: Add a warning message when unknown zone
```
Comment (kindermoumoute) [inline utils/locality.go]: yes

File: api/marketplace/v1/marketplace_utils.go [hunk 12]
```diff
@@ -2,46 +2,77 @@ package marketplace
...
+// or an error in case the image doesn't have a public version.
+func (image *Image) getLatestVersion() (*Version, error) {
+
+	for _, version := range image.Versions {
+		if version.Id == image.CurrentPublicVersion {
+			return version, nil
+		}
+	}
+
+	return nil, fmt.Errorf("no available version could be found for the image")
```
Comment (jerome-quere) [inline api/marketplace/v1/marketplace_utils.go]: latest version could be found for image %s, image.Name

**Pull request merged**